### PR TITLE
Add pr in build feature

### PR DESCRIPTION
**Pull request by ashwindasr**
Feature to check if a PR has been included in a build ([ART-1877](https://issues.redhat.com/browse/ART-1877)).

Workflow:
- Find the commits including and after the merge commit.
- Check each commit from the oldest to newest if has an associated successful build in the database, and report the first successful build.
- If no successful builds, repeat the above steps, but instead look for a failed build, and report the first failed build.
- If no builds have been found, report back saying that a build hasn't started yet for the PR.

### CODE REVIEW

File: artbotlib/pr_in_build.py [hunk 1]
```diff
@@ -205,6 +206,54 @@ def pr_merge_commit(self):
...
+            failed_builds = self.build_from_commit("failure")
+            if failed_builds:
+                first_failure = failed_builds[0]
+                self.logger.info(f"First failed build: {first_failure}")
+                self.so.say(
+                    f"No successful build found. First failed build: <https://brewweb.engineering.redhat.com/brew/buildinfo?buildID={first_failure}|{first_failure}>")
+            else:
+                self.logger.info(f"No builds have run yet.")
+                self.so.say("No builds have started yet for the PR. Check again later.")
+
```
Comment (locriandev) [inline artbotlib/pr_in_build.py:256]:
```suggestion
            return
        
        self.logger.info("No successful builds found given PR")
        failed_builds = self.build_from_commit("failure")
        if failed_builds:
            first_failure = failed_builds[0]
            self.logger.info(f"First failed build: {first_failure}")
            self.so.say(
                f"No successful build found. First failed build: <https://brewweb.engineering.redhat.com/brew/buildinfo?buildID={first_failure}|{first_failure}>")
            return
            
        self.logger.info(f"No builds have run yet.")
        self.so.say("No builds have started yet for the PR. Check again later.")

```

Comment (locriandev) [reply artbotlib/pr_in_build.py:256]: Suggesting this change to increase readability and keep indentantion level reasonable

**Review by locriandev — CHANGES_REQUESTED** [review]

File: artbotlib/pr_in_build.py [hunk 2]
```diff
@@ -205,6 +206,54 @@ def pr_merge_commit(self):
...
+                builds = response_data["results"]
+                build_ids = [x["build_0_id"] for x in builds]
+                return sorted(build_ids)
+
+    def find_builds(self):
+        """
+        Find successful or failed builds for the PR/merge commit. If none, report back to the user that the build
+        hasn't started yet.
+        """
+        successful_builds = self.build_from_commit("success")
```
[the remark this answers is not in the document]
Comment (ashwindasr) [reply artbotlib/pr_in_build.py]: Done! https://github.com/openshift/art-bot/pull/36/commits/cebf8969b7e558897d15ce0fc7cbf3a92e51a62b

File: artbotlib/pr_in_build.py (hunk 1 above)
Comment (ashwindasr) [reply artbotlib/pr_in_build.py]: Fixed https://github.com/openshift/art-bot/pull/36/commits/cebf8969b7e558897d15ce0fc7cbf3a92e51a62b

**Pull request merged**